File: leetcode/cpp/0167_twosum_ii.cpp
// O(n)
//derived from 0001_twosum

class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        unordered_map<int, int> record;

        for (int i = 0; i < numbers.size(); ++i) {
            auto item = record.find(target - numbers[i]);
            if (item != record.end()){
                return {item->second + 1, i + 1};
            }
            record[numbers[i]] = i;
        }
        return {};
    }
};

File: leetcode/python/0167_twosum_ii.py
# O(n) derived from 0001_twosum.py

class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        d = dict()

        for k, v in enumerate(numbers):
            if target - v in d:
                return [d[target - v] + 1, k + 1]
            d[v] = k
